LXDBinaryNotFound when install Kubernetes via conjure-up

## Report

When I had been installing Kubernetes via conjure-up, 'LXDBinaryNotFound' occurred!

I followed the link below
https://tutorials.ubuntu.com/tutorial/install-kubernetes-with-conjure-up?_ga=2.138719795.1250529933.1529720748-507910406.1527611129#1

![image](https://user-images.githubusercontent.com/35550209/41807300-15b65e3a-7708-11e8-9139-1b8743d523de.png)
![image](https://user-images.githubusercontent.com/35550209/41807353-f77075cc-7708-11e8-8e90-2f8cee8b3416.png)
![image](https://user-images.githubusercontent.com/35550209/41807355-fed48a24-7708-11e8-9720-b0b534c6a9e7.png)

## Please provide the output of the following commands

```
rokka@rokka-Lenovo-G50-30 ~ $ which juju
/snap/bin/juju
rokka@rokka-Lenovo-G50-30 ~ $ juju version
2.3.8-genericlinux-amd64
rokka@rokka-Lenovo-G50-30 ~ $ which conjure-up
/snap/bin/conjure-up
rokka@rokka-Lenovo-G50-30 ~ $ conjure-up --version
conjure-up 2.5.8
rokka@rokka-Lenovo-G50-30 ~ $ which lxc
rokka@rokka-Lenovo-G50-30 ~ $ /snap/bin/lxc config show
bash: /snap/bin/lxc: No such file or directory
rokka@rokka-Lenovo-G50-30 ~ $ /snap/bin/lxc version
bash: /snap/bin/lxc: No such file or directory
rokka@rokka-Lenovo-G50-30 ~ $ cat /etc/lsb-release
DISTRIB_ID=LinuxMint
DISTRIB_RELEASE=18.3
DISTRIB_CODENAME=sylvia
DISTRIB_DESCRIPTION="Linux Mint 18.3 Sylvia"
```

Please attach tarball of **~/.cache/conjure-up**:

```
tar cvzf conjure-up.tar.gz ~/.cache/conjure-up
```
[conjure-up.tar.gz](https://github.com/conjure-up/conjure-up/files/2129757/conjure-up.tar.gz)

## Sosreport

Please attach a sosreport:

```
rokka@rokka-Lenovo-G50-30 ~ $ sudo apt install sosreport
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  sosreport
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 118 kB of archives.
After this operation, 1,442 kB of additional disk space will be used.
Get:1 http://linux.yz.yamagata-u.ac.jp/ubuntu xenial-updates/main amd64 sosreport amd64 3.5-1~ubuntu16.04.3 [118 kB]
Fetched 118 kB in 0s (751 kB/s)   
Selecting previously unselected package sosreport.
(Reading database ... 294534 files and directories currently installed.)
Preparing to unpack .../sosreport_3.5-1~ubuntu16.04.3_amd64.deb ...
Unpacking sosreport (3.5-1~ubuntu16.04.3) ...
Processing triggers for man-db (2.7.5-1) ...
Setting up sosreport (3.5-1~ubuntu16.04.3) ...
rokka@rokka-Lenovo-G50-30 ~ $ sosreport

sosreport (version 3.5)

no valid plugins were enabled
```

## What Spell was Selected?
Nothing

## What provider (aws, maas, localhost, etc)?
Occurred when installed Kubernetes, so ...

### MAAS Users
No

## Commands ran
```
rokka@rokka-Lenovo-G50-30 ~ $ sudo apt install snapd -y
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  snapd
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 13.5 MB of archives.
After this operation, 79.9 MB of additional disk space will be used.
Get:1 http://linux.yz.yamagata-u.ac.jp/ubuntu xenial-updates/main amd64 snapd amd64 2.32.9 [13.5 MB]
Fetched 13.5 MB in 3s (3,761 kB/s)
Selecting previously unselected package snapd.
(Reading database ... 294738 files and directories currently installed.)
Preparing to unpack .../snapd_2.32.9_amd64.deb ...
Unpacking snapd (2.32.9) ...
Processing triggers for man-db (2.7.5-1) ...
Setting up snapd (2.32.9) ...
Job for snapd.seeded.service failed because the control process exited with error code. See "systemctl status snapd.seeded.service" and "journalctl -xe" for details.
snapd.seeded.service couldn't start.
snapd.snap-repair.service is a disabled or a static unit, not starting it.
```
```
rokka@rokka-Lenovo-G50-30 ~ $ sudo snap install conjure-up --classic
2018-06-23T17:36:54+09:00 INFO Waiting for restart...
conjure-up 2.5.7-20180606.1806 from 'canonical' installed
```

## Additional Information
```
rokka@rokka-Lenovo-G50-30 ~ $ cat /home/rokka/.cache/conjure-up/conjure-up.log
2018-06-23 16:39:20,526 [DEBUG] conjure-up/canonical-kubernetes - events.py:52 - Setting Shutdown at conjureup/events.py:145
2018-06-23 16:39:20,539 [DEBUG] conjure-up/canonical-kubernetes - events.py:52 - Received Shutdown at conjureup/events.py:176
2018-06-23 16:39:20,540 [INFO] conjure-up/canonical-kubernetes - events.py:180 - Shutting down
2018-06-23 16:39:27,611 [DEBUG] conjure-up/_unspecified_spell - juju.py:35 - bin_path candidate found
2018-06-23 16:39:27,612 [DEBUG] conjure-up/_unspecified_spell - juju.py:35 - wait_path candidate found
2018-06-23 16:39:29,490 [DEBUG] conjure-up/_unspecified_spell - app.py:279 - Juju version: 2.3.8-genericlinux-amd64, conjure-up version: 2.5.8
2018-06-23 16:39:29,550 [DEBUG] conjure-up/_unspecified_spell - telemetry.py:17 - Showing screen: Application Start
2018-06-23 16:39:29,570 [DEBUG] conjure-up/_unspecified_spell - telemetry.py:31 - OS: Linux-4.13.0-45-generic-x86_64-with-debian-stretch-sid 
2018-06-23 16:39:29,584 [INFO] conjure-up/_unspecified_spell - events.py:173 - Watching for shutdown
2018-06-23 16:39:29,854 [DEBUG] conjure-up/_unspecified_spell - events.py:52 - Awaiting Shutdown at conjureup/events.py:176
2018-06-23 16:39:29,938 [DEBUG] conjure-up/_unspecified_spell - telemetry.py:17 - Showing screen: Spell Selection
2018-06-23 16:39:38,524 [DEBUG] conjure-up/_unspecified_spell - telemetry.py:31 - Spell Choice: canonical-kubernetes 
2018-06-23 16:39:38,547 [DEBUG] conjure-up/canonical-kubernetes - download.py:60 - Path is local filesystem, copying /snap/conjure-up/1013/spells/canonical-kubernetes to /home/rokka/.cache/conjure-up/canonical-kubernetes
2018-06-23 16:39:41,709 [DEBUG] conjure-up/canonical-kubernetes - step.py:27 - steps: [<StepModel Canonical Distribution of Kubernetes 00_process-providertype v: False c: []>, <StepModel Canonical Distribution of Kubernetes 01_select-network v: True c: []>, <StepModel Canonical Distribution of Kubernetes 02_get-kubectl v: True c: []>, <StepModel Canonical Distribution of Kubernetes 03_cluster-info v: True c: []>, <StepModel Canonical Distribution of Kubernetes 04_enable-cni v: False c: []>]
2018-06-23 16:39:45,725 [DEBUG] conjure-up/canonical-kubernetes - __init__.py:126 - Pulling bundle for canonical-kubernetes from channel: stable
2018-06-23 16:39:48,556 [DEBUG] conjure-up/canonical-kubernetes - telemetry.py:17 - Showing screen: Add-on Selection
2018-06-23 16:39:56,510 [DEBUG] conjure-up/canonical-kubernetes - gui.py:78 - Starting watcher for verifying LXD server is available.
2018-06-23 16:39:56,511 [DEBUG] conjure-up/canonical-kubernetes - telemetry.py:17 - Showing screen: Choose a Cloud
2018-06-23 16:39:56,529 [DEBUG] conjure-up/canonical-kubernetes - events.py:52 - Setting Error at conjureup/events.py:149
2018-06-23 16:39:56,531 [ERROR] conjure-up/canonical-kubernetes - events.py:161 - Unhandled exception in <Task finished coro=<CloudsController._monitor_localhost() done, defined at /snap/conjure-up/1013/lib/python3.6/site-packages/conjureup/controllers/juju/clouds/gui.py:85> exception=AttributeError("module 'conjureup.errors' has no attribute 'LXDBinaryNotFound'",)>
Traceback (most recent call last):
  File "/snap/conjure-up/1013/lib/python3.6/site-packages/conjureup/controllers/juju/clouds/gui.py", line 92, in _monitor_localhost
    provider._set_lxd_dir_env()
  File "/snap/conjure-up/1013/lib/python3.6/site-packages/conjureup/models/provider.py", line 353, in _set_lxd_dir_env
    raise errors.LXDBinaryNotFound()
AttributeError: module 'conjureup.errors' has no attribute 'LXDBinaryNotFound'
2018-06-23 16:39:56,560 [DEBUG] conjure-up/canonical-kubernetes - __init__.py:29 - Showing dialog for exception: module 'conjureup.errors' has no attribute 'LXDBinaryNotFound'
```

--- a/conjureup/models/provider.py
+++ b/conjureup/models/provider.py
@@ -350,7 +350,7 @@ class Localhost(BaseProvider):
             app.env['LXD_DIR'] = str(self.lxd_socket_dir)
             self.lxc_bin = '/usr/bin/lxc'
         else:
-            raise errors.LXDBinaryNotFound()
+            raise errors.LXDBinaryNotFoundError()
         app.log.debug("LXD environment set: binary {} lxd_dir {}".format(
             self.lxc_bin, self.lxd_socket_dir))
 

